src/rust/catnip/runtime/mod.rs

```diff
@@ -438,22 +435,25 @@ impl NetworkRuntime for SharedDPDKRuntime {
                 assert!(header_size + body.len() >= MIN_PAYLOAD_SIZE);
 
                 // We're only using the header_mbuf for, well, the header.
-                header_mbuf.trim(header_mbuf.len() - header_size).unwrap();
+                header_mbuf.trim(header_mbuf.len() - header_size)?;
 
                 // Get the body mbuf.
                 let body_mbuf: *mut rte_mbuf = if body.is_dpdk_allocated() {
                     // The body is already stored in an MBuf, just extract it from the DemiBuffer.
-                    expect_some!(body.into_mbuf(), "'body' should be DPDK-allocated")
+                    match body.into_mbuf() {
+                        Some(body) => body,
+                        _ => return Err(Fail::new(libc::EINVAL, "'body' should be DPDK-allocated")),
+                    }
```

Comment:
In my opinion, this should really be an `expect_some!()`. If a wrongly allocated buffer made all its way down here, something really bad happend.

Reply:
sounds good